src/backend/gpopt/translate/CTranslatorScalarToDXL.cpp

```diff
@@ -1345,6 +1345,8 @@ CTranslatorScalarToDXL::TranslateFuncExprToDXL
 	const IMDFunction *md_func = m_md_accessor->RetrieveFunc(mdid_func);
 	if (IMDFunction::EfsVolatile == md_func->GetFuncStability())
 	{
+		m_context->m_has_volatile_functions = true;
```

Comment:
The definition of `m_context` states it may be `NULL`. Although this does not seem to be the case in `TranslateFuncExprToDXL`, I would propose to add a `NULL` check here.

Reply:
@leskin-in thanks for this remark!
@Stolb27 complained in a ticket about segfaults in `CTranslatorScalarToDXL.cpp`. According to the trace they happened on translation of table columns with default values (I suppose - functions) to DXL. May be this fix solve this problem. @Stolb27 can you run your tests again on updated version?